### PR TITLE
LL-2135 Integrate new CounterValues API

### PR DESCRIPTION
**Pull request by JunichiSugiura**
### Type

Feature

### Context

https://ledgerhq.atlassian.net/browse/LL-2135

Fixes https://github.com/LedgerHQ/ledger-live-desktop/issues/3122
Fixes https://github.com/LedgerHQ/ledger-live-desktop/issues/3250
Fixes https://github.com/LedgerHQ/ledger-live-desktop/issues/2326
Fixes https://github.com/LedgerHQ/ledger-live-desktop/issues/3279
Fixes https://github.com/LedgerHQ/ledger-live-desktop/issues/3236

### Parts of the app affected / Test plan

- Countervalue price for each amount Rows
- Portfolio Chart
- Account Chart
- Assets allocations related components
- Settings -> Crypto assets
- Settings -> General -> Countervalue
- Send flow -> Amount field

### CODE REVIEW

**Comment by JunichiSugiura** [comment]
### TODO
- [ ] [Extra polling logic](https://github.com/LedgerHQ/ledger-live-desktop/blob/0ac7738b2c59c94c4a031c3de354e15c5a2764dd/src/renderer/components/CountervaluesProvider.js#L43)

**Comment by gre** [comment]
@JunichiSugiura  let's try to sync this PR with latest develop of LLD and latest master+PR work of live-common so we can offer some builds for testing (even if we miss some small features)

File: src/renderer/components/CountervaluesProvider.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  useEffect(() => {
+    async function getSavedState() {
+      const values = await getKey("app", "countervalues");
+      setSavedState(values);
+    }
+    getSavedState();
+  }, []);
+
+  return (
+    <Countervalues userSettings={{ trackingPairs, autofillGaps: true }} savedState={savedState}>
```
Comment (gre) [inline src/renderer/components/CountervaluesProvider.js:27]: a memo might help for the userSettings (to check)

File: src/renderer/components/Price.js [hunk 2]
```diff
@@ -67,7 +46,20 @@ const Price = ({
...
-}: Props) => {
+}: Props) {
+  const effectiveUnit = unit || from.units[0];
+  const value = BigNumber(10 ** effectiveUnit.magnitude);
+  const rawCounterValueCurrency = useSelector(counterValueCurrencySelector);
+  const counterValueCurrency = to || rawCounterValueCurrency;
+  const rawCounterValue = useCalculate({
+    from,
+    to: counterValueCurrency,
+    value: value.toNumber(),
```
Comment (gre) [inline src/renderer/components/Price.js:57]: we don't need to use a BigNumber then?

**Review by gre — APPROVED** [review]
The work done here is very impressive. good job 🚀

File: src/renderer/components/Price.js (hunk 2 above)
Comment (JunichiSugiura) [reply src/renderer/components/Price.js]: BigNumber value is passed to CounterValue below.

CounterValue component doesn't really need to accept BigNumber as value prop but it's a way it is currently. Some component passes account.amount: BigNumber to it. I need to understand when to use BigNumber and when not to.

**Comment by gre** [comment]
@JunichiSugiura check the conflicts. not that bad it seems 🤞

**Review by Arnaud97234 — APPROVED** [review]

**Comment by jensamunch** [comment]
@gre this is still broken in 2.17.1?

image

**Comment by gre** [comment]
This is not yet in production. Will be in 2.18.0

**Pull request merged**